### PR TITLE
Decode the signature armor before trying to validate it

### PR DESCRIPTION
**Pull request by noxer**
#### Summary
Newer signatures files are "armored" my default. This PR adds a function to decode this armor before trying to validate the signature of an upgrade.

#### Ticket Link
[MM-46455](https://mattermost.atlassian.net/browse/MM-46455)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by pfltdv** [comment]
@noxer had same error with the change.
```
mattermost_1  | {"timestamp":"2022-08-18 15:43:00.955 Z","level":"debug","msg":"Unable to verify the Mattermost file signature","caller":"upgrader/upgrader_linux.go:129","error":"openpgp: invalid data: tag byte does not have MSB set"}
mattermost_1  | {"timestamp":"2022-08-18 15:43:00.955 Z","level":"error","msg":"Unable to verify the signature of the downloaded file","caller":"upgrader/upgrader_linux.go:262","error":"invalid file signature"}
```

File: services/upgrader/upgrader_linux.go [hunk 1]
```diff
@@ -108,21 +109,51 @@ func verifySignature(filename string, sigfilename string, publicKey []byte) erro
...
+	defer signature.Close()
 
-	_, err = openpgp.CheckDetachedSignature(keyring, mattermost_tar, signature)
+	decodedSignature, err := decodeArmoredSignature(signature)
+	if err != nil {
+		mlog.Debug("Unable to decode the Mattermost .sig file to verify the file signature", mlog.Err(err))
+		return NewInvalidSignature()
+	}
+
+	_, err = openpgp.CheckDetachedSignature(keyring, mattermost_tar, decodedSignature)
```
Comment (pfltdv) [inline services/upgrader/upgrader_linux.go:127]: Do we need to execute `CheckDetachedSignature` when `decodeArmoredSignature` is successful ?

If error is nil for `decodeArmoredSignature`, `CheckDetachedSignature` will generate error.

Comment (noxer) [reply services/upgrader/upgrader_linux.go:127]: Yes, `decodeArmoredSignature` only sets up a decoder for the armored block, it doesn't validate the signature, it doesn't even validate that the complete signature can be decoded.

**Comment by noxer** [comment]
It seems like this was not the root cause of our validation issues.

**Comment by isacikgoz** [comment]
> It seems like this was not the root cause of our validation issues.

@noxer Is this still a valid PR that we should proceed?

**Comment by noxer** [comment]
@isacikgoz I think we can proceed. While this doesn't fix the root cause, it allows us to ve more flexible with the format of the signature file.

**Comment by agnivade** [comment]
If it does not solve any specific problem, I am leaning towards leaving it as is. But I will defer to @pfltdv / @phoinixgrr if they think it's helpful. No big issues from my side.

**Comment by pfltdv** [comment]
Since it does not solve our main problem, better to keep it as it is. Since we may need to change our library which does not need any decoding.

**Comment by noxer** [comment]
Fine by me. Let's close it.

**Pull request closed without merging**